omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PasswordResource.java

```diff
@@ -0,0 +1,96 @@
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.wrapper.openmrs1_8.UserAndPassword1_8;
+import org.openmrs.util.PrivilegeConstants;
+
+@Resource(name = RestConstants.VERSION_1 + "/password", supportedClass = UserAndPassword1_8.class, supportedOpenmrsVersions = { "2.0.*,2.1.*" })
+public class PasswordResource extends DelegatingCrudResource<UserAndPassword1_8> {
+
+	@Override
+	public UserAndPassword1_8 newDelegate() {
+		return new UserAndPassword1_8();
+	}
+	
+	@Override
+	public Object create(SimpleObject propertiesToCreate, RequestContext context) throws ResponseException {
+		User authenticatedUser = Context.getAuthenticatedUser();
+		if(authenticatedUser != null){
+		  return update(authenticatedUser.getUuid(), propertiesToCreate, context);
+		}
+		else{
+			throw new APIAuthenticationException("You are not authenticated to change password");
+		}
+	}
+	
+	@Override
+	public UserAndPassword1_8 save(UserAndPassword1_8 user) {
+		User openmrsUser = user.getUser();
+		if (openmrsUser == null || openmrsUser.getUserId() == null) {
+			throw new APIAuthenticationException("User not available");
+		} else {
+			//only authenticated users can change their own password or other's password
+			User authenticatedUser = Context.getAuthenticatedUser();
+			if (authenticatedUser.hasPrivilege(PrivilegeConstants.EDIT_USER_PASSWORDS)) {
+				Context.getUserService().changePassword(openmrsUser, user.getPassword());
+			} else if (authenticatedUser.equals(openmrsUser)) {
+				Context.getUserService().changePassword(user.getOldPassword(), user.getPassword());
+			} else {
+				throw new APIAuthenticationException("You doesn't have privilege to change others password");
+			}
+		}
+		return new UserAndPassword1_8(openmrsUser);
+	}
+
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("password");
+		description.addProperty("oldPassword");
+
+		return description;
+	}
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		return null;
+	}
+
+	@Override
+	public UserAndPassword1_8 getByUniqueId(String uuid) {
+		Context.addProxyPrivilege(PrivilegeConstants.GET_USERS);
+		UserAndPassword1_8 userAndPassword1_8 = new UserAndPassword1_8(Context.getUserService().getUserByUuid(uuid));
+		Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
+		return userAndPassword1_8;
```

Comment:
wrap this in a try/finally block, e.g. so you still remove the privilege if an exception is thrown. I.e.

    try {
        addProxyPrivilege(...);
        // do something, return, etc
    } finally {
        removeProxyPrvilege(...)
    }